sklearn/utils/estimator_checks.py

```diff
@@ -1408,6 +1439,11 @@ def check_class_weight_classifiers(name, classifier_orig):
         X, y = make_blobs(centers=n_centers, random_state=0, cluster_std=20)
         X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=.5,
                                                             random_state=0)
+
+        X_train = gram_matrix_if_pairwise(X_train, classifier_orig,
+                                          kernel=rbf_kernel)
+        X_test = gram_matrix_if_pairwise(X_test, classifier_orig,
```

Comment:
This surely can't work. The test data needs to be the kernel applied on X_test with respect to the training.

Reply:
It's a wonder that tests are passing